### ISSUE TITLE
`[Parameter] must be integers` even when I set the space to Integer only

### ISSUE THREAD

**Issue by NTNguyen13** [scikit-optimize/scikit-optimize#738]
In my code, I tried to perform Hyperparameter Optimization according to this: https://scikit-optimize.github.io/notebooks/hyperparameter-optimization.html

```

space = [Integer(10, 100, name='min_cluster_size'),
         Integer(10, 100, name='min_samples')
         ]

import hdbscan
h_dbscan = hdbscan.HDBSCAN(core_dist_n_jobs=40, memory='/home/cngc3/BMC/BMC/')

@use_named_args(space)
def objective(**params):
    h_dbscan.set_params(**params)
    h_dbscan_cluster = h_dbscan.fit_predict(embedded_umap)
    a = metrics.adjusted_rand_score(drop_label, np.delete(h_dbscan_cluster, index, axis=0))

    return -a

from skopt import gp_minimize
res_gp = gp_minimize(objective, space, n_calls=50, random_state=0)
"Best score=%.4f" % res_gp.fun
```

the parameter 'min_samples' and 'min_dist' should have Integer value, so I set the space to use the Integer from `skopt.space`. But I received the error:

```
Traceback (most recent call last):
  File "/home/cngc3/anaconda3/envs/CBC/lib/python3.6/site-packages/IPython/core/interactiveshell.py", line 2961, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-14-3fcfbf4f1f3d>", line 3, in <module>
    res_gp = gp_minimize(objective, space, n_calls=50, random_state=0)
  File "/home/cngc3/anaconda3/envs/CBC/lib/python3.6/site-packages/skopt/optimizer/gp.py", line 228, in gp_minimize
    callback=callback, n_jobs=n_jobs)
  File "/home/cngc3/anaconda3/envs/CBC/lib/python3.6/site-packages/skopt/optimizer/base.py", line 248, in base_minimize
    next_y = func(next_x)
  File "/home/cngc3/anaconda3/envs/CBC/lib/python3.6/site-packages/skopt/utils.py", line 636, in wrapper
    objective_value = func(**arg_dict)
  File "<ipython-input-13-c1fd9102ce4e>", line 12, in objective
    h_dbscan_cluster = h_dbscan.fit_predict(embedded_umap)
  File "/home/cngc3/anaconda3/envs/CBC/lib/python3.6/site-packages/hdbscan/hdbscan_.py", line 876, in fit_predict
    self.fit(X)
  File "/home/cngc3/anaconda3/envs/CBC/lib/python3.6/site-packages/hdbscan/hdbscan_.py", line 854, in fit
    self._min_spanning_tree) = hdbscan(X, **kwargs)
  File "/home/cngc3/anaconda3/envs/CBC/lib/python3.6/site-packages/hdbscan/hdbscan_.py", line 456, in hdbscan
    raise ValueError('Min samples and min cluster size must be integers!')
ValueError: Min samples and min cluster size must be integers!
```

I have checked by using only 1 hyperparameter at a time in space, but I still got the problem.

Any help is welcome, thank you

**Comment by NTNguyen13** [comment]
I changed the space to use Categorical instead of Integer:

```

space = [Categorical(list(range(10,100)), name='cluster_size'),
         Categorical(list(range(10,100)), name='samples')
         ]

default_parameters = [65, 80]
@use_named_args(dimensions=space)
def objective(samples, cluster_size):
    h_dbscan = hdbscan.HDBSCAN(min_samples=samples, min_cluster_size=cluster_size,
                               core_dist_n_jobs=40,
                               memory='/home/cngc3/BMC/BMC/')
    print('Min sample size:', samples)
    print('Min cluster size:', cluster_size)

    h_dbscan_cluster = h_dbscan.fit_predict(embedded_umap)
    a = metrics.adjusted_rand_score(drop_label, np.delete(h_dbscan_cluster, index, axis=0))
    print('ARI:', a)
    print('==================================')
    return -a

res_gp = gp_minimize(func=objective, dimensions=space, n_calls=200, random_state=0,x0=default_parameters)
"Best score=%.4f" % res_gp.fun
```

it runs fine for the first 11 epoch, and then the same error:` ValueError: Min samples and min cluster size must be integers!`. At that time, the sample size and cluster size were both 57. I used different random state and found out that it will stop at epoch 12, no matter what values were used.

I think it's the same error with this issue: https://github.com/scikit-optimize/scikit-optimize/issues/715
, skopt using wrong values after a while, so the estimators raise errors

**Comment by iaroslav-ai** [comment]
This seems like it might also be an issue with the type of values that skopt suggests; Do you use the latest pip version of the package, or the `master` branch of the repo?

**Comment by NTNguyen13** [comment]
@iaroslav-ai hi, I'm using the version from conda forge: https://anaconda.org/conda-forge/scikit-optimize

**Comment by jolespin** [comment]
Was this ever resolved? I'm getting the same situation with hdbscan (v0.8.24?) and skopt (v0.7).

**Comment by holgern** [comment]
This issue seems to me the same as issue #790
This issue is currently tried to solved in #807

**Comment by jolespin** [comment]
I ended up using int(x) and float(x) within my func to force the types of my parameters which fixed it.

**Comment by holgern** [comment]
Should be fixed in 0.7.1 when using:
```
space = [Integer(10, 100, name='min_cluster_size', dtype=int),
         Integer(10, 100, name='min_samples', dtype=int)
         ]
```
